### PR TITLE
c8d/image/inspect: Return `Descriptor`

### PR DESCRIPTION
**Pull request by vvoland**
- follow up to: https://github.com/moby/moby/pull/48861

Add the image target descriptor to the image inspect response.

**- How to verify it**
```bash
$ docker pull alpine
$ curl -s --unix-socket /var/run/docker.sock localhost/images/alpine/json | jq
{
  "Id": "sha256:1e42bbe2508154c9126d48c2b8a75420c3544343bf86fd041fb7527e017a4b4a",
  "RepoTags": [
    "alpine:latest"
  ],
  "RepoDigests": [
    "alpine@sha256:1e42bbe2508154c9126d48c2b8a75420c3544343bf86fd041fb7527e017a4b4a"
  ],
  "Parent": "",
  "Comment": "buildkit.dockerfile.v0",
  "Created": "2024-09-06T12:05:36Z",
...
  "Descriptor": {
    "mediaType": "application/vnd.oci.image.index.v1+json",
    "digest": "sha256:1e42bbe2508154c9126d48c2b8a75420c3544343bf86fd041fb7527e017a4b4a",
    "size": 9218
  }
}
```
```bash
$ docker pull arm64v8/alpine:3.10

$ curl -s --unix-socket /var/run/docker.sock localhost/images/arm64v8/alpine:3.10/json | jq

{
  "Id": "sha256:f73072b1ae04902f4d7511b834a1ea2d0475ddf6f87bd81f06e2b4590974a2e5",
  "RepoTags": [
    "arm64v8/alpine:3.10"
  ],
  "RepoDigests": [
    "arm64v8/alpine@sha256:f73072b1ae04902f4d7511b834a1ea2d0475ddf6f87bd81f06e2b4590974a2e5"
  ],
  "Parent": "",
  "Comment": "",
  "Created": "2021-04-14T18:43:17.827903143Z",
...
  "Descriptor": {
    "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
    "digest": "sha256:f73072b1ae04902f4d7511b834a1ea2d0475ddf6f87bd81f06e2b4590974a2e5",
    "size": 528
  }

```

**- Description for the changelog**

```markdown changelog
# Merge with the `GET /images/json`
containerd image store: `GET /images/json` and `GET /images/{name}/json` response now includes `Descriptor` field, which contains an OCI descriptor of the image target. The new field will only be populated if the daemon provides a multi-platform image store.
```

**- A picture of a cute animal (not mandatory but encouraged)**
Untitled

### CODE REVIEW

File: docs/api/version-history.md [hunk 1]
```diff
@@ -34,8 +34,8 @@ keywords: "API, Docker, rcli, REST, documentation"
   and will be omitted in API v1.49.
 * `Sysctls` in `HostConfig` (top level `--sysctl` settings) for `eth0` are
   no longer migrated to `DriverOpts`, as described in the changes for v1.46.
-* `GET /images/json` response now includes `Descriptor` field, which contains
-  an OCI descriptor of the image target.
+* `GET /images/json` and `GET /images/{name}/json` response now includes 
+  `Descriptor` field, which contains an OCI descriptor of the image target.
```
Comment (thaJeztah) [inline docs/api/version-history.md:38]:
```suggestion
* `GET /images/json` and `GET /images/{name}/json` response now include a
  `Descriptor` field, which contains an OCI descriptor of the image target.
```

**Review by thaJeztah — APPROVED** [review]
LGTM

should we have an integration test for this?

**Comment by vvoland** [comment]
Added a simple integration test

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by laurazard — APPROVED** [review]
LGTM

**Pull request merged**